arm-cognitiveservices/2016-02-01-preview/swagger/cognitiveservices.json

```diff
@@ -0,0 +1,691 @@
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type"
+        }
+      }
+    },
+    "CognitiveServicesAccountListResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CognitiveServicesAccount"
+          },
+          "description": "Gets the list of Cognitive Services accounts and their properties."
+        }
+      },
+      "description": "The list of cognitive services accounts operation response."
+    },
+    "CognitiveServicesAccountProperties": {
+      "properties": {
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Gets the status of the cognitive services account at the time the operation was called.",
+          "enum": [
+            "Creating",
+            "ResolvingDNS",
+            "Succeeded",
+            "Failed"
+          ],
+          "x-ms-enum": {
+            "name": "ProvisioningState",
+            "modelAsString": false
+          }
+        },
+        "endpoint": {
+          "type": "string",
+          "description": "Endpoint of the created account"
+        }
+      }
+    },
+    "CognitiveServicesAccountKeys": {
+      "properties": {
+        "key1": {
+          "type": "string",
+          "description": "Gets the value of key 1."
+        },
+        "key2": {
+          "type": "string",
+          "description": "Gets the value of key 2."
+        }
+      },
+      "description": "The access keys for the cognitive services account."
+    },
+    "RegenerateKeyParameters": {
+      "properties": {
+        "keyName": {
+          "type": "string",
+          "description": "key name to generate (key1|key2)",
+          "enum": [
+            "Key1",
+            "Key2"
+          ],
+          "x-ms-enum": {
+            "name": "KeyNameEnum",
+            "modelAsString": false
+          }
+        }
+      },
+      "description": "Regenerate key parameters."
+    },
+    "CognitiveServicesAccountEnumerateSkusResult": {
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CognitiveServicesResourceAndSku"
+          },
+          "description": "Gets the list of Cognitive Services accounts and their properties."
+        }
+      },
+      "description": "The list of cognitive services accounts operation response."
+    },
+    "CognitiveServicesResourceAndSku": {
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "Resource Type"
```

Comment:
Should this be an enum. I expect a handful of resource types. You can always mark "x-ms-enum" extension's modelAsString property as false. This will generate a static class with constant properties as enum values. So it gives intellisense to customers.